whois-commons/src/main/java/net/ripe/db/whois/common/hazelcast/WhoisHazelcastBlockedIps.java

```diff
@@ -0,0 +1,64 @@
+package net.ripe.db.whois.common.hazelcast;
+
+import com.google.common.base.Joiner;
+import com.hazelcast.collection.ISet;
+import com.hazelcast.core.HazelcastInstance;
+import net.ripe.db.whois.common.ip.IpInterval;
+import net.ripe.db.whois.common.profiles.WhoisProfile;
+import org.eclipse.jetty.util.StringUtil;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Component
+@Profile({WhoisProfile.DEPLOYED})
+public class WhoisHazelcastBlockedIps implements HazelcastBlockedIps{
+
+    private static final Logger LOGGER = getLogger(WhoisHazelcastBlockedIps.class);
+
+    private static final Joiner COMMA_JOINER = Joiner.on(',');
+
+    private final ISet<IpInterval> ipBlockedSet;
+
+    public WhoisHazelcastBlockedIps(final HazelcastInstance hazelcastInstance,
+                                    @Value("${ipranges.blocked.list:}") final String blockedListIps) {
+
+        ipBlockedSet = hazelcastInstance.getSet("ipBlockedSet");
+
+        for (final String address : StringUtil.csvSplit(blockedListIps)) {
+            addBlockedListAddress(address);
+        }
+
+        LOGGER.info("hazelcast instances {} members: {} " , hazelcastInstance.getName() , hazelcastInstance.getCluster().getMembers());
+    }
+
+    @Override
+    public void addBlockedListAddress(final String address) {
+        ipBlockedSet.add(IpInterval.parse(address));
+        LOGGER.info("Ipaddress {} added to blocked list", address);
+    }
+
+    @Override
+    public void removeBlockedListAddress(String address) {
+        ipBlockedSet.remove(IpInterval.parse(address));
+        LOGGER.info("Ipaddress {} removed from blocked list", address);
+    }
+
+    @Override
+    public String getBlockedList() {
+        StringBuilder result = new StringBuilder();
+        COMMA_JOINER.appendTo(result, ipBlockedSet);
+
+        LOGGER.info("The blocked list contains next IPs {}", result);
+        return String.format("The blocked list contains next IPs %s", result);
```

Comment:
Just return the comma separated string if its needed, not with the message

Reply:
It is not needed I am adding this just because I like that the call returns a pretty well formed message.

Reply:
Looking at message name feels like you are sending blocked Ips as a string

Reply:
and because probably in the future you want a way to see if your change has been applied.
I think that this is ok

Reply:
In that case shift the message to JMX call and logging is also fine.